src/nvim/tui/tui.c

```diff
@@ -69,6 +70,9 @@ typedef struct {
   bool can_use_terminal_scroll;
   bool mouse_enabled;
   bool busy;
+  /* use int instead ? */
+  // cursorentry_T cursor_shapes[SHAPE_IDX_COUNT]; /**!< is it fine to reuse cursorentry_T here ? */
+  int cursor_shapes[SHAPE_IDX_COUNT]; /**!< is it fine to reuse cursorentry_T here ? */
```

Comment:
At first in tui_cursor_style_set, I converted the data back to cursorentry_T as in nvim core, hence the `cursorentry_T cursor_shapes[SHAPE_IDX_COUNT]` but then I thought it might better to generate directly the control sequence characters ?